### ISSUE TITLE
Change Java/Scala package names

### ISSUE THREAD

**Issue by gmethvin** [playframework/playframework#5816]
For Play 3.0.0 we should separate the APIs in some clean way. I'm not sure there's a consensus on this yet, but `play.japi` and `play.sapi` seem like like the most popular options.

**Comment by schmitch** [comment]
wouldn't it be better to just use play.java.api? japi or sapi sound's aweful.

**Comment by gmethvin** [comment]
The problem with `play.java.api` is the way scala relative imports work. You can have package name clashes with `java` and `scala`.

I'm not too sure about `japi`/`sapi` either. Maybe `javaapi`/`scalaapi`?

**Comment by schmitch** [comment]
Hm and what about play.j.api and play.s.api? would be the same as sapi, but since there would be a . between . between s/j and api

**Comment by gmethvin** [comment]
Yeah, that might work. We still have time to discuss since this is something we'll probably do all at once right before the 3.0.0 release.

**Comment by schmitch** [comment]
@gmethvin it would've been also great to have a draft doc about what's coming in play 3.0 so that the community could take a look at that?!

**Comment by gmethvin** [comment]
We have a [roadmap](https://docs.google.com/document/d/11sVi1-REAIDFVHvwBrfRt1uXkBzROHQYgmcZNGJtDnA) but that actually needs updating. There will likely be a 2.6 release to ease in the transition to 3.0.

Not all the things I'm putting on github for 3.0.0 are completely thought out, but I wanted to get all of it up here to make it easier for the community to contribute.

**Comment by zzguiji** [comment]
prefer `play.scala.api` and `play.java.api`

**Comment by schmitch** [comment]
Actually akka and lagom are using `javadsl` and `scaladsl`. just as a note.

**Comment by kflorence** [comment]
my vote is for `play.scala.api` and `play.java.api` as well. i don't really understand the issue with relative imports.

**Comment by fmeriaux** [comment]
Yes, javadsl & scaladsl is very good 👍

**Comment by hepin1989** [comment]
@fmeriaux Yes,I vote to alias with akka's package name,just `javadsl` and `scaladsl`.

**Comment by nafg** [comment]
It's not a DSL
